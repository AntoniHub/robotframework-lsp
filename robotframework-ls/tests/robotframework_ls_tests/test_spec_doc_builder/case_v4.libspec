<?xml version="1.0" encoding="UTF-8"?>
<keywordspec name="case_argspec" type="LIBRARY" format="ROBOT" scope="GLOBAL" generated="2022-02-16T17:26:13Z" specversion="4" source="case_argspec.py" lineno="1">
<version/>
<doc>Documentation for library ``case_argspec``.</doc>
<tags>
</tags>
<inits>
</inits>
<keywords>
<kw name="Arg String Default" lineno="42">
<arguments repr="arg1=my:,my:something,*foo,**foo,&lt;rara&gt;">
<arg kind="POSITIONAL_OR_NAMED" required="false" repr="arg1=my:,my:something,*foo,**foo,&lt;rara&gt;">
<name>arg1</name>
<default>my:,my:something,*foo,**foo,&lt;rara&gt;</default>
</arg>
</arguments>
<doc/>
<shortdoc/>
</kw>
<kw name="Arg With Enum Type" lineno="26">
<arguments repr="model: SomeEnum">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="model: SomeEnum">
<name>model</name>
<type>SomeEnum</type>
</arg>
</arguments>
<doc/>
<shortdoc/>
</kw>
<kw name="Arg With Enum Type And Default" lineno="30">
<arguments repr="model: SomeEnum = v2">
<arg kind="POSITIONAL_OR_NAMED" required="false" repr="model: SomeEnum = v2">
<name>model</name>
<type>SomeEnum</type>
<default>v2</default>
</arg>
</arguments>
<doc/>
<shortdoc/>
</kw>
<kw name="Arg With Positional" lineno="38">
<arguments repr="arg1, *, arg2">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="arg1">
<name>arg1</name>
</arg>
<arg kind="NAMED_ONLY_MARKER" required="false" repr="*">
</arg>
<arg kind="NAMED_ONLY" required="true" repr="arg2">
<name>arg2</name>
</arg>
</arguments>
<doc/>
<shortdoc/>
</kw>
<kw name="Arg With Starargs" lineno="34">
<arguments repr="arg1, arg2=10, *arg3, **arg4">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="arg1">
<name>arg1</name>
</arg>
<arg kind="POSITIONAL_OR_NAMED" required="false" repr="arg2=10">
<name>arg2</name>
<default>10</default>
</arg>
<arg kind="VAR_POSITIONAL" required="false" repr="*arg3">
<name>arg3</name>
</arg>
<arg kind="VAR_NAMED" required="false" repr="**arg4">
<name>arg4</name>
</arg>
</arguments>
<doc/>
<shortdoc/>
</kw>
<kw name="Arg With Type" lineno="20">
<arguments repr="model: int">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="model: int">
<name>model</name>
<type>int</type>
</arg>
</arguments>
<doc>Some doc in arg_with_type.</doc>
<shortdoc>Some doc in arg_with_type.</shortdoc>
</kw>
<kw name="Simple Arg" lineno="14">
<arguments repr="model">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="model">
<name>model</name>
</arg>
</arguments>
<doc>Some doc in simple_arg.</doc>
<shortdoc>Some doc in simple_arg.</shortdoc>
</kw>
</keywords>
<datatypes>
<enums>
<enum name="SomeEnum">
<doc>An enumeration.</doc>
<members>
<member name="v1" value="0"/>
<member name="v2" value="1"/>
</members>
</enum>
</enums>
</datatypes>
</keywordspec>
