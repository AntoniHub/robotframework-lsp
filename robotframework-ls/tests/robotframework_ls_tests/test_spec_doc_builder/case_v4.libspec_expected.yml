Arg String Default:
  args:
  - arg_type: null
    default_value: my:,my:something,*foo,**foo,<rara>
    is_keyword: false
    is_star: false
    name: arg1
    original_arg: arg1=my:,my:something,*foo,**foo,<rara>
Arg With Enum Type:
  args:
  - arg_type: SomeEnum
    default_value: null
    is_keyword: false
    is_star: false
    name: model
    original_arg: 'model: SomeEnum'
Arg With Enum Type And Default:
  args:
  - arg_type: SomeEnum
    default_value: v2
    is_keyword: false
    is_star: false
    name: model
    original_arg: 'model: SomeEnum = v2'
Arg With Positional:
  args:
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: false
    name: arg1
    original_arg: arg1
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: false
    name: arg2
    original_arg: arg2
Arg With Starargs:
  args:
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: false
    name: arg1
    original_arg: arg1
  - arg_type: null
    default_value: '10'
    is_keyword: false
    is_star: false
    name: arg2
    original_arg: arg2=10
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: true
    name: '@arg3'
    original_arg: '*arg3'
  - arg_type: null
    default_value: null
    is_keyword: true
    is_star: false
    name: '&arg4'
    original_arg: '**arg4'
Arg With Type:
  args:
  - arg_type: int
    default_value: null
    is_keyword: false
    is_star: false
    name: model
    original_arg: 'model: int'
Simple Arg:
  args:
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: false
    name: model
    original_arg: model
data_types:
- doc: An enumeration.
  members:
  - name: v1
    value: '0'
  - name: v2
    value: '1'
  name: SomeEnum
  type: Enum
